app/server/utils/transfer_limits.py

```diff
@@ -0,0 +1,251 @@
+
+
+def is_liquid_and_user_is_kyc_business_verified(credit_transfer):
+    if is_user_and_liquid_token(credit_transfer):
+        return False
+
+    return user_business_kyc_is_verified(credit_transfer)
+
+# ~~~~~~LIMIT FILTERS~~~~~~
+
+
+@curry
+def after_time_period_filter(days: int, query: Query):
+
+    epoch = datetime.datetime.today() - datetime.timedelta(days=days)
+
+    return query.filter(CreditTransfer.created >= epoch)
+
+
+@curry
+def matching_sender_user_filter(transfer: CreditTransfer, query: Query):
+    return query.filter(CreditTransfer.sender_user == transfer.sender_user)
+
+
+@curry
+def matching_transfer_type_filter(transfer: CreditTransfer, query: Query,):
+    return query.filter(CreditTransfer.transfer_type == transfer.transfer_type)
+
+
+@curry
+def matching_transfer_type_and_subtype_filter(transfer: CreditTransfer, query: Query):
+    return (query
+            .filter(CreditTransfer.transfer_type == transfer.transfer_type)
+            .filter(CreditTransfer.transfer_subtype == transfer.transfer_subtype))
+
+
+@curry
+def withdrawal_or_agent_out_filter(transfer: CreditTransfer, query: Query):
+    return query.filter(or_(CreditTransfer.transfer_type == TransferTypeEnum.WITHDRAWAL,
+                            CreditTransfer.transfer_subtype == TransferSubTypeEnum.AGENT_OUT))
+
+
+@curry
+def empty_filter(transfer: CreditTransfer, query: Query):
+    return query
+
+
+class TransferLimit(object):
+
+    def __repr__(self):
+        return f"<TransferLimit {self.name}>"
+
+    def apply_all_filters(self, transfer: CreditTransfer, query: Query):
+
+        return pipe(query,
+                    matching_sender_user_filter(transfer),
+                    after_time_period_filter(self.time_period_days),
+                    self.transfer_filter(transfer))
+
+    def __init__(self,
+                 name: str,
+                 applied_to_transfer_types: AppliedToTypes,
+                 application_filter: Callable,
+                 time_period_days: int,
+                 total_amount: Optional[int] = None,
+                 transfer_count: Optional[int] = None,
+                 transfer_filter: Optional[Query.filter] = matching_transfer_type_filter,
+                 transfer_balance_fraction: Optional[float] = None):
+
+        self.name = name
+
+        # Force to list of tuples to ensure the use of 'in' behaves as expected
+        self.applied_to_transfer_types = [tuple(t) if isinstance(t, list) else t for t in applied_to_transfer_types]
+
+        self.application_filter = application_filter
+        self.time_period_days = time_period_days
+        self.total_amount = total_amount
+        self.transfer_count = transfer_count
+        self.transfer_filter = transfer_filter
+        self.transfer_balance_fraction = transfer_balance_fraction
+
+
+LIMITS = [
+    TransferLimit('Sempo Level 0: P7', [STANDARD_PAYMENT], is_liquid_and_user_is_not_phone_and_not_kyc_verified, 7,
```

Comment:
I extended the names so that we could filter on them during tests - will this have any negative side effect @tristanhcole ?

Reply:
Other than the string comparison in the tests file, no